test/index.test.ts

```diff
@@ -0,0 +1,197 @@
+  let app: App;
+
+  beforeEach(function() {
+    // set up stuff
+    app = new App({
+      id: APP_ID,
+      privateKey: PRIVATE_KEY
+    });
+  });
+
+  // see https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app
+  it("gets bearer token", function() {
+    const bearer = app.getSignedJsonWebToken();
+    expect(bearer).toEqual(BEARER);
+  });
+
+  // see https://developer.github.com/v3/apps/#create-a-new-installation-token
+  it("gets installation token", () => {
+    nock("https://api.github.com", {
+      reqheaders: {
+        authorization: `bearer ${BEARER}` // installation access token
+      }
+    })
+      .post("/app/installations/123/access_tokens")
+      .reply(201, {
+        token: "foo"
+      });
+
+    return app
+      .getInstallationAccessToken({ installationId: 123 })
+      .then(token => {
+        expect(token).toEqual("foo");
+      });
+  });
+
+  it("gets installation token from cache", () => {
+    nock("https://api.github.com")
+      .post("/app/installations/123/access_tokens")
+      .reply(201, {
+        token: "foo"
+      });
+
+    return app
+      .getInstallationAccessToken({ installationId: 123 })
+      .then(token => {
+        expect(token).toEqual("foo");
+
+        return app.getInstallationAccessToken({ installationId: 123 });
+      })
+      .then(token => {
+        expect(token).toEqual("foo");
+      });
+  });
+
+  it("caches based on installation id", () => {
+    nock("https://api.github.com")
+      .post("/app/installations/123/access_tokens")
+      .reply(201, {
+        token: "foo"
+      })
+      .post("/app/installations/456/access_tokens")
+      .reply(201, {
+        token: "bar"
+      });
+
+    return app
+      .getInstallationAccessToken({ installationId: 123 })
+      .then(token => {
+        expect(token).toEqual("foo");
+
+        return app.getInstallationAccessToken({ installationId: 456 });
+      })
+      .then(token => {
+        expect(token).toEqual("bar");
+      });
+  });
+
+  const oneHourInMs = 1000 * 60 * 60;
+  it("request installation again after timeout", () => {
+    const mock = nock("https://api.github.com")
+      .post("/app/installations/123/access_tokens")
+      .reply(201, {
+        token: "foo"
+      })
+      .post("/app/installations/123/access_tokens")
+      .reply(201, {
+        token: "bar"
+      });
+
+    return app
+      .getInstallationAccessToken({ installationId: 123 })
+      .then(token => {
+        expect(token).toEqual("foo");
+
+        return new Promise(resolve => {
+          setTimeout(resolve, oneHourInMs);
+          clock.tick(oneHourInMs);
+        });
+      })
+      .then(() => {
+        return app.getInstallationAccessToken({ installationId: 123 });
+      })
+      .then(token => {
+        expect(token).toEqual("bar");
+        expect(mock.pendingMocks()).toStrictEqual([]);
+      });
+  }); //.timeout(oneHourInMs + 2000);
```

Comment:
I think we can just remove the `//.timeout(oneHourInMs + 2000);` comment. I think with mocha it just with a timeout error because of the `clock.tick(oneHourInMs);` line. If it no longer does that, we’r good